### PR TITLE
Update dependencies and to 2020f

### LINKED ISSUE #27: Figure out JDK 11-13 for android CI
Temporarily disabled in #26

### PR DESCRIPTION
**Pull request by ZacSweers**
* Update dependencies (Mostly kotlin)
* iana 2020f
* add Kotlin ABI validator for compiler artifact

Resolves #27 and some others along the way

### CODE REVIEW

**Comment by ZacSweers** [comment]
The auto-updater failed for some reason, but will look into that separately

**Comment by ZacSweers** [comment]
I don't know what killed the emulator

**Comment by ZacSweers** [comment]
Filed an issue here: https://github.com/ReactiveCircus/android-emulator-runner/issues/118

**Comment by gabrielittner** [comment]
Will review this tomorrow

File: build.gradle [hunk 1]
```diff
@@ -13,23 +13,22 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import kotlin.KotlinVersion
-
 buildscript {
   dependencies {
     // Needs to go first, else gradle's forced 1.3.72 version takes precedence
-    classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:1.4.10")
+    classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:1.4.20")
```
Comment (gabrielittner) [inline build.gradle:19]: There is Kotlin 1.4.21

File: ticktock-gradle-plugin/api/ticktock-gradle-plugin.api [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+public abstract class dev/zacsweers/ticktock/gradle/TickTockExtension {
+	public abstract fun getCodeOutputDir ()Lorg/gradle/api/file/DirectoryProperty;
+	public final fun getLanguage ()Lorg/gradle/api/provider/Property;
+	public final fun getPackageName ()Lorg/gradle/api/provider/Property;
+	public final fun getTzOutputDir ()Lorg/gradle/api/file/DirectoryProperty;
+	public final fun getTzVersion ()Lorg/gradle/api/provider/Property;
+}
+
+public final class dev/zacsweers/ticktock/gradle/TickTockPlugin : org/gradle/api/Plugin {
+	public static final field INTERMEDIATES Ljava/lang/String;
```
Comment (gabrielittner) [inline ticktock-gradle-plugin/api/ticktock-gradle-plugin.api:25]: This probably shouldn't be public.

**Review by gabrielittner — APPROVED** [review]
I'm wondering if all the removed Australia zones will lead to issues on Android

File: ticktock-gradle-plugin/api/ticktock-gradle-plugin.api (hunk 2 above)
Comment (ZacSweers) [reply ticktock-gradle-plugin/api/ticktock-gradle-plugin.api:25]: Good catch, will tidy up

**Comment by ZacSweers** [comment]
Let me regenerate just in case

File: ticktock-gradle-plugin/api/ticktock-gradle-plugin.api (hunk 2 above)
Comment (ZacSweers) [reply ticktock-gradle-plugin/api/ticktock-gradle-plugin.api]: ef0474e

File: build.gradle (hunk 1 above)
Comment (ZacSweers) [reply build.gradle]: https://github.com/ZacSweers/ticktock/pull/46/commits/dd1d450a491aeee13411a375d60c535f340b8aab

**Pull request merged**